<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>MAFDash: R package to easily create an HTML dashboard to summarize and visualize data from Mutation Annotation Format (MAF) file • MAFDash</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="MAFDash: R package to easily create an HTML dashboard to summarize and visualize data from Mutation Annotation Format (MAF) file">
<meta property="og:description" content="MAFDash">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MAFDash</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.2.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/MAFDash.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Quick_Start.html">Quick Start</a>
    </li>
    <li>
      <a href="../articles/Customizing.html">Customizing</a>
    </li>
    <li>
      <a href="../articles/Advanced_Example.html">Advanced Example</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/ashishjain1988/MAFDash/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="MAFDash_files/header-attrs-2.9/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>MAFDash: R package to easily create an HTML dashboard to summarize and visualize data from Mutation Annotation Format (MAF) file</h1>
                        <h4 class="author"><div class="line-block">Mayank Tandon, Ashish Jain<br>
Frederick National Laboratory For Cancer Research<br>
Frederick, Maryland<br>
</div></h4>
            
            <h4 class="date">01/25/2022</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/ashishjain1988/MAFDash/blob/master/vignettes/MAFDash.Rmd"><code>vignettes/MAFDash.Rmd</code></a></small>
      <div class="hidden name"><code>MAFDash.Rmd</code></div>

    </div>

    
    
<!-- ``` -->
<!-- #{r setup, include = FALSE} -->
<!-- knitr::opts_chunk$set( -->
<!--   collapse = TRUE, -->
<!--   comment = "#>", -->
<!--   echo = TRUE, -->
<!--   message = FALSE, -->
<!--   warning = FALSE -->
<!-- ) -->
<!-- ``` -->
<div id="mafdash" class="section level1">
<h1 class="hasAnchor">
<a href="#mafdash" class="anchor"></a>MAFDash</h1>
<div id="view-examples-on-github-io" class="section level4">
<h4 class="hasAnchor">
<a href="#view-examples-on-github-io" class="anchor"></a>View examples on <a href="https://mtandon09.github.io/MAFDash/">github.io</a>
</h4>
</div>
</div>
<div id="how-to-get-help-for-mafdash" class="section level1">
<h1 class="hasAnchor">
<a href="#how-to-get-help-for-mafdash" class="anchor"></a>How to get help for MAFDash</h1>
<p>Please post all the questions or queries related to MAFDash package on <strong>Github Issues</strong>. This will help us to build an information repository which can be used by other users.</p>
<p><a href="https://github.com/ashishjain1988/MAFDash/issues">https://github.com/ashishjain1988/MAFDash/issues</a></p>
</div>
<div id="getting-started" class="section level1">
<h1 class="hasAnchor">
<a href="#getting-started" class="anchor"></a>Getting started</h1>
<div id="installation-from-cran" class="section level2">
<h2 class="hasAnchor">
<a href="#installation-from-cran" class="anchor"></a>Installation from CRAN</h2>
<p><code><a href="https://rdrr.io/r/utils/install.packages.html">install.packages("MAFDash")</a></code></p>
</div>
<div id="installation-from-github" class="section level2">
<h2 class="hasAnchor">
<a href="#installation-from-github" class="anchor"></a>Installation from Github</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"dplyr"</span>,<span class="st">"ensurer"</span>,<span class="st">"ggplot2"</span>,<span class="st">"tidyr"</span>,<span class="st">"DT"</span>,<span class="st">"rmarkdown"</span>,<span class="st">"knitr"</span>,<span class="st">"flexdashboard"</span>,<span class="st">"htmltools"</span>,<span class="st">"data.table"</span>,<span class="st">"ggbeeswarm"</span>,<span class="st">"RColorBrewer"</span>,<span class="st">"plotly"</span>,<span class="st">"circlize"</span>,<span class="st">"canvasXpress"</span>,<span class="st">"crosstalk"</span>,<span class="st">"bsplus"</span>,<span class="st">"BiocManager"</span>,<span class="st">"maftools"</span>,<span class="st">"ComplexHeatmap"</span><span class="op">)</span><span class="op">)</span>
<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"TCGAbiolinks"</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="va">devtools</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://devtools.r-lib.org/">devtools</a></span><span class="op">)</span>
<span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org/reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">"ashishjain1988/MAFDashRPackage"</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="scope" class="section level1">
<h1 class="hasAnchor">
<a href="#scope" class="anchor"></a>Scope</h1>
<p><a href="https://docs.gdc.cancer.gov/Encyclopedia/pages/Mutation_Annotation_Format/">Mutation Annotation Format (MAF)</a> is a tabular data format used for storing genetic mutation data. For example, <a href="https://www.cancer.gov/about-nci/organization/ccg/research/structural-genomics/tcga">The Cancer Genome Atlas (TCGA)</a> project has made MAF files from each project publicly available.</p>
<p>The package – <strong>MAFDash</strong> – contains a set of R tools to easily create an HTML dashboard to summarize and visualize data from MAF file.</p>
<p>The resulting HTML file serves as a self-contained report that can be used to explore the result. Currently, MAFDash produces mostly static plots powered by <a href="https://bioconductor.org/packages/release/bioc/vignettes/maftools/inst/doc/maftools.html">maftools</a>, <a href="https://github.com/jokergoo/ComplexHeatmap">ComplexHeatmap</a> and <a href="https://github.com/jokergoo/circlize">circlize</a>, as well as interactive visualizations using <a href="https://cran.r-project.org/package=canvasXpress">canvasXpress</a> and <a href="https://plotly.com/r/">plotly</a>. The report is generated with a parameterized <a href="https://rmarkdown.rstudio.com/">R Markdown</a> script that uses <a href="https://rmarkdown.rstudio.com/flexdashboard/">flexdashboard</a> to arrange all the information.</p>
<p>This package is a companion to the Shiny app, <a href="https://github.com/mtandon09/mafwiz">MAFWiz</a>. Instead of relying on a Shiny server, this dashboard is an attempt to try some of those things using client-side javascript functionality.</p>
</div>
<div id="make-the-dashboard" class="section level1">
<h1 class="hasAnchor">
<a href="#make-the-dashboard" class="anchor"></a>Make the dashboard</h1>
<p><a href="https://docs.gdc.cancer.gov/Encyclopedia/pages/Mutation_Annotation_Format/">Mutation Annotation Format (MAF)</a> is a tabular data format used for storing genetic mutation data. For example, <a href="https://www.cancer.gov/about-nci/organization/ccg/research/structural-genomics/tcga">The Cancer Genome Atlas (TCGA)</a> project has made MAF files from each project publicly available. The main function of MAFDash (<code>getMAFDashboard</code>) creates an HTML dashboard to summarize and visualize data from MAF files. The resulting HTML file serves as a self-contained report that can be used to explore and share the results. The example below shows how we can create an HTML MAF dashboard file. The first argument of <code>getMAFDashboard</code> can be anything that’s accepted by maftools’s <a href="https://rdrr.io/bioc/maftools/man/read.maf.html"><code>read.maf</code></a> function (path to a file, or a <code>MAF</code> , <code>data.frame</code>, or <code>data.table</code> object)</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/ashishjain1988/MAFDash/">MAFDash</a></span><span class="op">)</span></code></pre></div>
<pre><code>## Loading required package: rmarkdown</code></pre>
<pre><code>## Loading required package: knitr</code></pre>
<pre><code>## Loading required package: htmltools</code></pre>
<pre><code>## Loading required package: ggplot2</code></pre>
<pre><code>## Loading required package: TCGAbiolinks</code></pre>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">maf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"test.mutect2.maf.gz"</span>, package <span class="op">=</span> <span class="st">"MAFDash"</span><span class="op">)</span>
<span class="fu"><a href="../reference/getMAFDashboard.html">getMAFDashboard</a></span><span class="op">(</span><span class="va">maf</span>, outputFileName<span class="op">=</span><span class="st">"output"</span>, outputFileTitle<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"MAF Dashboard - Test"</span><span class="op">)</span>,outputFilePath <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html">tempdir</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## processing file: MAFDash.Rmd</code></pre>
<pre><code>## Warning in DT::datatable(summaryTable, style = "bootstrap4", class =
## c("display"), : `autoHideNavigation` will be ignored if the `pageLength` option
## is not provided.</code></pre>
<pre><code>## All mutation types: Missense Mutation, Frame Shift Del, Nonsense
## Mutation, Splice Site, Multi Hit, In Frame Del, Frame Shift Ins.</code></pre>
<pre><code>## `alter_fun` is assumed vectorizable. If it does not generate correct
## plot, please set `alter_fun_is_vectorized = FALSE` in `oncoPrint()`.</code></pre>
<pre><code>## Warning in generateVariantTable(maf): Not outputting these columns: gnomAD_AF,
## AF</code></pre>
<pre><code>## output file: /private/var/folders/jy/45byld4s14q9wc9qvbwx027m7rhngt/T/RtmpneQlA5/file163ce4376e51/MAFDash.knit.md</code></pre>
<pre><code>## /usr/local/bin/pandoc +RTS -K512m -RTS /private/var/folders/jy/45byld4s14q9wc9qvbwx027m7rhngt/T/RtmpneQlA5/file163ce4376e51/MAFDash.knit.md --to html4 --from markdown+autolink_bare_uris+tex_math_single_backslash --output /private/var/folders/jy/45byld4s14q9wc9qvbwx027m7rhngt/T/RtmpneQlA5/output.html --lua-filter /Users/jaina13/Library/R/4.0/library/rmarkdown/rmarkdown/lua/pagebreak.lua --lua-filter /Users/jaina13/Library/R/4.0/library/rmarkdown/rmarkdown/lua/latex-div.lua --self-contained --variable bs3=TRUE --standalone --section-divs --template /Users/jaina13/Library/R/4.0/library/flexdashboard/rmarkdown/templates/flex_dashboard/resources/default.html --include-in-header /var/folders/jy/45byld4s14q9wc9qvbwx027m7rhngt/T//RtmpneQlA5/rmarkdown-str163ce532b8a00.html --variable theme=yeti --include-in-header /var/folders/jy/45byld4s14q9wc9qvbwx027m7rhngt/T//RtmpneQlA5/rmarkdown-str163ce50809411.html --mathjax --variable 'mathjax-url:https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML' --include-in-header /var/folders/jy/45byld4s14q9wc9qvbwx027m7rhngt/T//RtmpneQlA5/file163ce29791161html --include-after-body /var/folders/jy/45byld4s14q9wc9qvbwx027m7rhngt/T//RtmpneQlA5/file163ce6afa3759.html --highlight-style pygments --include-before-body /var/folders/jy/45byld4s14q9wc9qvbwx027m7rhngt/T//RtmpneQlA5/file163ce4b640c8c.html --include-after-body /var/folders/jy/45byld4s14q9wc9qvbwx027m7rhngt/T//RtmpneQlA5/file163ce1573e1ff.html</code></pre>
<pre><code>## 
## Output created: /private/var/folders/jy/45byld4s14q9wc9qvbwx027m7rhngt/T/RtmpneQlA5/output.html</code></pre>
<div id="example" class="section level2">
<h2 class="hasAnchor">
<a href="#example" class="anchor"></a>Example</h2>
<p>Here are some example dashboards created using TCGA data: - <a href="https://mtandon09.github.io/MAFDash/output/TCGA-UVM.MAFDash.html">TCGA-UVM</a> - <a href="https://mtandon09.github.io/MAFDash/output/TCGA-BRCA.MAFDash.html">TCGA-BRCA</a></p>
</div>
</div>
<div id="downloading-tcga-mutation-data-in-maf-format" class="section level1">
<h1 class="hasAnchor">
<a href="#downloading-tcga-mutation-data-in-maf-format" class="anchor"></a>Downloading TCGA mutation data in MAF format</h1>
<p>MAFDash also provides a wrapper function <code>getMAFdataTCGA</code> around the <code>TCGABiolinks</code>, which returns the mutation data of different cancers in MAF format from TCGA website. See <a href="https://gdc.cancer.gov/resources-tcga-users/tcga-code-tables/tcga-study-abbreviations">this page</a> for a list of TCGA codes.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/ashishjain1988/MAFDash/">"MAFDash"</a></span><span class="op">)</span>
<span class="co"># Download MAF data from TCGA</span>
<span class="va">CancerCode</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"ACC"</span>,<span class="st">"UVM"</span><span class="op">)</span>
<span class="va">inputFolderPath</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/tempfile.html">tempdir</a></span><span class="op">(</span><span class="op">)</span>,<span class="st">"/tcga_data"</span><span class="op">)</span> <span class="co">## This folder will be created if it doesn't exist </span>
<span class="va">maf</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getMAFdataTCGA.html">getMAFdataTCGA</a></span><span class="op">(</span>cancerCode <span class="op">=</span> <span class="va">CancerCode</span>, outputFolder <span class="op">=</span> <span class="va">inputFolderPath</span><span class="op">)</span></code></pre></div>
<p>#Creating individual plots using the MAF dataset</p>
<p>##Oncoplot The oncoplot shows the number and types of mutations in a set of genes across the samples. The function <code>generateOncoPlot</code> can be used to generate the oncoplot.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/ashishjain1988/MAFDash/">MAFDash</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/PoisonAlien/maftools">maftools</a></span><span class="op">)</span>
<span class="va">maf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"test.mutect2.maf.gz"</span>, package <span class="op">=</span> <span class="st">"MAFDash"</span><span class="op">)</span>
<span class="fu"><a href="../reference/generateOncoPlot.html">generateOncoPlot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/maftools/man/read.maf.html">read.maf</a></span><span class="op">(</span><span class="va">maf</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## -Reading
## -Validating
## -Silent variants: 561 
## -Summarizing
## --Possible FLAGS among top ten genes:
##   MACF1
##   MUC16
## -Processing clinical data
## --Missing clinical data
## -Finished in 0.222s elapsed (0.205s cpu)</code></pre>
<pre><code>## All mutation types: Missense Mutation, Frame Shift Del, Nonsense
## Mutation, Splice Site, Multi Hit, In Frame Del, Frame Shift Ins.</code></pre>
<pre><code>## `alter_fun` is assumed vectorizable. If it does not generate correct
## plot, please set `alter_fun_is_vectorized = FALSE` in `oncoPrint()`.</code></pre>
<p><img src="MAFDash_files/figure-html/unnamed-chunk-4-1.png" width="700"></p>
<p>##Burden Plot The burdenplot compares the total number of mutations between the samples using a dotplot. The figure also have a barplot showing the distribution of different type of mutations across the samples using a barplot.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/ashishjain1988/MAFDash/">MAFDash</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/PoisonAlien/maftools">maftools</a></span><span class="op">)</span>
<span class="va">maf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"test.mutect2.maf.gz"</span>, package <span class="op">=</span> <span class="st">"MAFDash"</span><span class="op">)</span>
<span class="fu"><a href="../reference/generateBurdenPlot.html">generateBurdenPlot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/maftools/man/read.maf.html">read.maf</a></span><span class="op">(</span><span class="va">maf</span><span class="op">)</span>, plotType<span class="op">=</span><span class="st">"Dotplot"</span><span class="op">)</span></code></pre></div>
<pre><code>## -Reading
## -Validating
## -Silent variants: 561 
## -Summarizing
## --Possible FLAGS among top ten genes:
##   MACF1
##   MUC16
## -Processing clinical data
## --Missing clinical data
## -Finished in 0.775s elapsed (0.747s cpu)</code></pre>
<p><img src="MAFDash_files/figure-html/unnamed-chunk-5-1.png" width="700"></p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/generateBurdenPlot.html">generateBurdenPlot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/maftools/man/read.maf.html">read.maf</a></span><span class="op">(</span><span class="va">maf</span><span class="op">)</span>, plotType<span class="op">=</span><span class="st">"Barplot"</span><span class="op">)</span></code></pre></div>
<pre><code>## -Reading
## -Validating
## -Silent variants: 561 
## -Summarizing
## --Possible FLAGS among top ten genes:
##   MACF1
##   MUC16
## -Processing clinical data
## --Missing clinical data
## -Finished in 0.237s elapsed (0.216s cpu)</code></pre>
<p><img src="MAFDash_files/figure-html/unnamed-chunk-5-2.png" width="700"></p>
<p>##Mutation Type plot This function generates silent and non-silent mutation plot using the MAF data.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/ashishjain1988/MAFDash/">MAFDash</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/PoisonAlien/maftools">maftools</a></span><span class="op">)</span>
<span class="va">maf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"test.mutect2.maf.gz"</span>, package <span class="op">=</span> <span class="st">"MAFDash"</span><span class="op">)</span>
<span class="fu"><a href="../reference/generateMutationTypePlot.html">generateMutationTypePlot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/maftools/man/read.maf.html">read.maf</a></span><span class="op">(</span><span class="va">maf</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## -Reading
## -Validating
## -Silent variants: 561 
## -Summarizing
## --Possible FLAGS among top ten genes:
##   MACF1
##   MUC16
## -Processing clinical data
## --Missing clinical data
## -Finished in 0.239s elapsed (0.221s cpu)</code></pre>
<pre><code>## Warning in data.table::dcast(silent_classif_data, Tumor_Sample_Barcode ~ : The
## dcast generic in data.table has been passed a grouped_df and will attempt to
## redirect to the reshape2::dcast; please note that reshape2 is deprecated, and
## this redirection is now deprecated as well. Please do this redirection yourself
## like reshape2::dcast(silent_classif_data). In the next version, this warning
## will become an error.</code></pre>
<p><img src="MAFDash_files/figure-html/unnamed-chunk-6-1.png" width="700"></p>
<p>##TiTv plot This function plot the frequency of Transitions and Transversions of gene mutations</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/ashishjain1988/MAFDash/">MAFDash</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/PoisonAlien/maftools">maftools</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://plotly-r.com">plotly</a></span><span class="op">)</span></code></pre></div>
<pre><code>## 
## Attaching package: 'plotly'</code></pre>
<pre><code>## The following object is masked from 'package:ggplot2':
## 
##     last_plot</code></pre>
<pre><code>## The following object is masked from 'package:stats':
## 
##     filter</code></pre>
<pre><code>## The following object is masked from 'package:graphics':
## 
##     layout</code></pre>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">maf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"test.mutect2.maf.gz"</span>, package <span class="op">=</span> <span class="st">"MAFDash"</span><span class="op">)</span>
<span class="va">plots</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/generateTiTvPlot.html">generateTiTvPlot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/maftools/man/read.maf.html">read.maf</a></span><span class="op">(</span><span class="va">maf</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## -Reading
## -Validating
## -Silent variants: 561 
## -Summarizing
## --Possible FLAGS among top ten genes:
##   MACF1
##   MUC16
## -Processing clinical data
## --Missing clinical data
## -Finished in 0.232s elapsed (0.215s cpu)</code></pre>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#plotly::subplot(plotly::subplot(plots$tiTvPatterns,plots$TiTv, nrows = 1, widths = c(1, 0.5)),plots$barplot,nrows = 2)</span></code></pre></div>
<p>##TCGA Compare plot This function plot the comparison of the mutation load against TCGA cohorts</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/ashishjain1988/MAFDash/">MAFDash</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/PoisonAlien/maftools">maftools</a></span><span class="op">)</span>
<span class="va">maf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"test.mutect2.maf.gz"</span>, package <span class="op">=</span> <span class="st">"MAFDash"</span><span class="op">)</span>
<span class="va">maf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/maftools/man/read.maf.html">read.maf</a></span><span class="op">(</span>maf <span class="op">=</span> <span class="va">maf</span><span class="op">)</span></code></pre></div>
<pre><code>## -Reading
## -Validating
## -Silent variants: 561 
## -Summarizing
## --Possible FLAGS among top ten genes:
##   MACF1
##   MUC16
## -Processing clinical data
## --Missing clinical data
## -Finished in 0.189s elapsed (0.178s cpu)</code></pre>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">l</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/generateTCGAComparePlot.html">generateTCGAComparePlot</a></span><span class="op">(</span>maf <span class="op">=</span> <span class="va">maf</span>, cohortName <span class="op">=</span> <span class="st">"test"</span><span class="op">)</span></code></pre></div>
<pre><code>## Warning in FUN(X[[i]], ...): Removed 0 samples with zero mutations.</code></pre>
<pre><code>## Performing pairwise t-test for differences in mutation burden..</code></pre>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">l</span><span class="op">$</span><span class="va">tcga_compare_plot</span></code></pre></div>
<p><img src="MAFDash_files/figure-html/unnamed-chunk-8-1.png" width="700"></p>
</div>
<div id="adding-custom-plots" class="section level1">
<h1 class="hasAnchor">
<a href="#adding-custom-plots" class="anchor"></a>Adding Custom Plots</h1>
<p>The <code><a href="../reference/getMAFDashboard.html">getMAFDashboard()</a></code> function will accept a named list for adding arbitrary objects to the dashboard. Each item in the list will be displayed in separate tabs, and the name of the element will be used as the title of the tab.</p>
<p>Elements of the list can be:</p>
<ul>
<li>
<strong>A boolean value</strong>: indicates whether or not to draw one of the preset plots provided by MAFDasoh</li>
<li>
<strong>A file path</strong>: any image format (or PDF)</li>
<li>
<strong>An R plot object</strong>: ggplot, plotly, ComplexHeatmap</li>
</ul>
<p>This functionality can be used with or without providing a MAF file. When MAF data is not provided, the “Variant Table” tab of the dashboard is automatically omitted.</p>
<div id="toy-example-with-iris-data" class="section level2">
<h2 class="hasAnchor">
<a href="#toy-example-with-iris-data" class="anchor"></a>Toy example with <code>iris</code> data</h2>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://plotly-r.com">plotly</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jokergoo/ComplexHeatmap">ComplexHeatmap</a></span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">iris</span><span class="op">)</span>

<span class="co">## Simple ggplot</span>
<span class="va">myplot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">iris</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">Sepal.Length</span>, y<span class="op">=</span><span class="va">Sepal.Width</span>, color<span class="op">=</span><span class="va">Species</span><span class="op">)</span><span class="op">)</span>

<span class="co">## Save as PNG (provide absolute file path)</span>
<span class="va">mycustomimage_png</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/getwd.html">getwd</a></span><span class="op">(</span><span class="op">)</span>,<span class="st">"custom_ggplot.png"</span><span class="op">)</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggsave.html">ggsave</a></span><span class="op">(</span><span class="va">mycustomimage_png</span>, plot<span class="op">=</span><span class="va">myplot</span>, width<span class="op">=</span><span class="fl">5</span>, height<span class="op">=</span><span class="fl">4</span><span class="op">)</span>

<span class="co">## Save as PDF (provide absolute file path)</span>
<span class="va">mycustomimage_pdf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/getwd.html">getwd</a></span><span class="op">(</span><span class="op">)</span>,<span class="st">"custom_ggplot.pdf"</span><span class="op">)</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggsave.html">ggsave</a></span><span class="op">(</span><span class="va">mycustomimage_pdf</span>, plot<span class="op">=</span><span class="va">myplot</span>, width<span class="op">=</span><span class="fl">5</span>, height<span class="op">=</span><span class="fl">4</span><span class="op">)</span>

<span class="co">## Convert ggplot to plotly</span>
<span class="va">myplotly</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/plotly/man/ggplotly.html">ggplotly</a></span><span class="op">(</span><span class="va">myplot</span><span class="op">)</span>

<span class="co">## Make heatmap with ComplexHeatmap</span>
<span class="va">hmdata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html">t</a></span><span class="op">(</span><span class="va">iris</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span><span class="op">)</span>
<span class="va">hmanno</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/HeatmapAnnotation.html">HeatmapAnnotation</a></span><span class="op">(</span>df<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>Species<span class="op">=</span><span class="va">iris</span><span class="op">[</span>,<span class="fl">5</span><span class="op">]</span><span class="op">)</span><span class="op">)</span>
<span class="va">myhm</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/Heatmap.html">Heatmap</a></span><span class="op">(</span><span class="va">hmdata</span>, bottom_annotation <span class="op">=</span> <span class="va">hmanno</span><span class="op">)</span>

<span class="co">## Customizable plotly from https://github.com/mtandon09/Dynamic_Plotly</span>
<span class="kw"><a href="https://rdrr.io/r/base/source.html">source</a></span><span class="op">(</span><span class="st">"https://raw.githubusercontent.com/mtandon09/Dynamic_Plotly/master/make_cutomizable_plotly.R"</span><span class="op">)</span>
<span class="va">custom_plotly</span> <span class="op">&lt;-</span> <span class="fu">make_customizable_plotly</span><span class="op">(</span><span class="va">iris</span><span class="op">)</span>

<span class="co">## Put together objects/filepaths into a list</span>
<span class="va">toyplotlist</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="st">"ggplot"</span><span class="op">=</span> <span class="va">myplot</span>,
                   <span class="st">"plotly"</span><span class="op">=</span> <span class="va">myplotly</span>,
                   <span class="st">"PNG"</span><span class="op">=</span> <span class="va">mycustomimage_png</span>,
                   <span class="st">"PDF"</span><span class="op">=</span> <span class="va">mycustomimage_pdf</span>,
                   <span class="st">"ComplexHeatmap"</span><span class="op">=</span> <span class="va">myhm</span>,
                   <span class="st">"Customizable"</span><span class="op">=</span> <span class="va">custom_plotly</span>
<span class="op">)</span>

<span class="co">## Filename to output to</span>
<span class="va">html_filename</span><span class="op">=</span><span class="st">"toy_dash.html"</span>

<span class="co">## Render dashboard</span>
<span class="fu"><a href="../reference/getMAFDashboard.html">getMAFDashboard</a></span><span class="op">(</span>plotList <span class="op">=</span> <span class="va">toyplotlist</span>,
                outputFileName <span class="op">=</span> <span class="va">html_filename</span>,
                outputFileTitle <span class="op">=</span> <span class="st">"Iris"</span><span class="op">)</span></code></pre></div>
<p><strong>Output</strong> The output can be seen <a href="https://mtandon09.github.io/MAFDashRPackage/examples/toy_dash.html.MAFDash.html">here</a>.</p>
<p>#Advanced Example</p>
</div>
<div id="download-tcga-data" class="section level2">
<h2 class="hasAnchor">
<a href="#download-tcga-data" class="anchor"></a>Download TCGA data</h2>
<div id="download-maf-file-using-tcgabiolinks" class="section level3">
<h3 class="hasAnchor">
<a href="#download-maf-file-using-tcgabiolinks" class="anchor"></a>Download MAF file using <code>TCGABiolinks</code>
</h3>
<p>MAFDash provides a wrapper function that tries to simplify retrieving data using <code>TCGABiolinks</code>. Valid project codes can be viewed by running <code>TCGABiolinks::getGDCprojects()</code> and checking the “tumor” column.</p>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/ashishjain1988/MAFDash/">MAFDash</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/BioinformaticsFMRP/TCGAbiolinks">TCGAbiolinks</a></span><span class="op">)</span>

<span class="va">tcga_code</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"ACC"</span>,<span class="st">"UVM"</span><span class="op">)</span>
<span class="co">#inputFolderPath &lt;- paste0(tempdir()) ## This folder will be created if it doesn't exist </span>
<span class="va">caller</span> <span class="op">=</span> <span class="st">"mutect2"</span>
<span class="va">title_label</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"TCGA-"</span>,<span class="va">tcga_code</span><span class="op">)</span>

<span class="va">maf_files</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getMAFdataTCGA.html">getMAFdataTCGA</a></span><span class="op">(</span><span class="va">tcga_code</span>,variant_caller <span class="op">=</span> <span class="va">caller</span><span class="op">)</span></code></pre></div>
<pre><code>## ============================================================================</code></pre>
<pre><code>##  For more information about MAF data please read the following GDC manual and web pages:</code></pre>
<pre><code>##  GDC manual: https://gdc-docs.nci.nih.gov/Data/PDF/Data_UG.pdf</code></pre>
<pre><code>##  https://gdc-docs.nci.nih.gov/Data/Bioinformatics_Pipelines/DNA_Seq_Variant_Calling_Pipeline/</code></pre>
<pre><code>##  https://gdc.cancer.gov/about-gdc/variant-calling-gdc</code></pre>
<pre><code>## ============================================================================</code></pre>
<pre><code>## --------------------------------------</code></pre>
<pre><code>## o GDCquery: Searching in GDC database</code></pre>
<pre><code>## --------------------------------------</code></pre>
<pre><code>## Genome of reference: hg38</code></pre>
<pre><code>## --------------------------------------------</code></pre>
<pre><code>## oo Accessing GDC. This might take a while...</code></pre>
<pre><code>## --------------------------------------------</code></pre>
<pre><code>## ooo Project: TCGA-ACC</code></pre>
<pre><code>## --------------------</code></pre>
<pre><code>## oo Filtering results</code></pre>
<pre><code>## --------------------</code></pre>
<pre><code>## ooo By access</code></pre>
<pre><code>## ooo By data.type</code></pre>
<pre><code>## ooo By workflow.type</code></pre>
<pre><code>## ----------------</code></pre>
<pre><code>## oo Checking data</code></pre>
<pre><code>## ----------------</code></pre>
<pre><code>## ooo Check if there are duplicated cases</code></pre>
<pre><code>## ooo Check if there results for the query</code></pre>
<pre><code>## -------------------</code></pre>
<pre><code>## o Preparing output</code></pre>
<pre><code>## -------------------</code></pre>
<pre><code>## Downloading data for project TCGA-ACC</code></pre>
<pre><code>## GDCdownload will download: 2.666716 MB</code></pre>
<pre><code>## Downloading as: TCGA.ACC.mutect.81ac2c46-37db-4dcd-923a-061a7ae626a3.DR-10.0.somatic.maf.gz</code></pre>
<pre><code>## 
  |                                                                            
  |                                                                      |   0%
  |                                                                            
  |                                                                      |   1%
  |                                                                            
  |=                                                                     |   1%
  |                                                                            
  |=                                                                     |   2%
  |                                                                            
  |==                                                                    |   2%
  |                                                                            
  |==                                                                    |   3%
  |                                                                            
  |===                                                                   |   4%
  |                                                                            
  |===                                                                   |   5%
  |                                                                            
  |====                                                                  |   5%
  |                                                                            
  |====                                                                  |   6%
  |                                                                            
  |=====                                                                 |   7%
  |                                                                            
  |=====                                                                 |   8%
  |                                                                            
  |======                                                                |   8%
  |                                                                            
  |======                                                                |   9%
  |                                                                            
  |=======                                                               |  10%
  |                                                                            
  |========                                                              |  11%
  |                                                                            
  |========                                                              |  12%
  |                                                                            
  |=========                                                             |  12%
  |                                                                            
  |=========                                                             |  13%
  |                                                                            
  |==========                                                            |  14%
  |                                                                            
  |==========                                                            |  15%
  |                                                                            
  |===========                                                           |  15%
  |                                                                            
  |===========                                                           |  16%
  |                                                                            
  |============                                                          |  17%
  |                                                                            
  |============                                                          |  18%
  |                                                                            
  |=============                                                         |  18%
  |                                                                            
  |=============                                                         |  19%
  |                                                                            
  |==============                                                        |  19%
  |                                                                            
  |==============                                                        |  20%
  |                                                                            
  |==============                                                        |  21%
  |                                                                            
  |===============                                                       |  21%
  |                                                                            
  |===============                                                       |  22%
  |                                                                            
  |================                                                      |  22%
  |                                                                            
  |================                                                      |  23%
  |                                                                            
  |=================                                                     |  24%
  |                                                                            
  |=================                                                     |  25%
  |                                                                            
  |==================                                                    |  26%
  |                                                                            
  |===================                                                   |  27%
  |                                                                            
  |===================                                                   |  28%
  |                                                                            
  |====================                                                  |  28%
  |                                                                            
  |====================                                                  |  29%
  |                                                                            
  |=====================                                                 |  30%
  |                                                                            
  |======================                                                |  31%
  |                                                                            
  |======================                                                |  32%
  |                                                                            
  |=======================                                               |  32%
  |                                                                            
  |=======================                                               |  33%
  |                                                                            
  |========================                                              |  34%
  |                                                                            
  |========================                                              |  35%
  |                                                                            
  |=========================                                             |  35%
  |                                                                            
  |=========================                                             |  36%
  |                                                                            
  |==========================                                            |  36%
  |                                                                            
  |==========================                                            |  37%
  |                                                                            
  |==========================                                            |  38%
  |                                                                            
  |===========================                                           |  38%
  |                                                                            
  |===========================                                           |  39%
  |                                                                            
  |============================                                          |  39%
  |                                                                            
  |============================                                          |  40%
  |                                                                            
  |============================                                          |  41%
  |                                                                            
  |=============================                                         |  41%
  |                                                                            
  |=============================                                         |  42%
  |                                                                            
  |==============================                                        |  42%
  |                                                                            
  |==============================                                        |  43%
  |                                                                            
  |===============================                                       |  44%
  |                                                                            
  |===============================                                       |  45%
  |                                                                            
  |================================                                      |  45%
  |                                                                            
  |================================                                      |  46%
  |                                                                            
  |=================================                                     |  47%
  |                                                                            
  |==================================                                    |  48%
  |                                                                            
  |==================================                                    |  49%
  |                                                                            
  |===================================                                   |  50%
  |                                                                            
  |===================================                                   |  51%
  |                                                                            
  |====================================                                  |  51%
  |                                                                            
  |====================================                                  |  52%
  |                                                                            
  |=====================================                                 |  52%
  |                                                                            
  |=====================================                                 |  53%
  |                                                                            
  |======================================                                |  54%
  |                                                                            
  |======================================                                |  55%
  |                                                                            
  |=======================================                               |  55%
  |                                                                            
  |=======================================                               |  56%
  |                                                                            
  |========================================                              |  57%
  |                                                                            
  |=========================================                             |  58%
  |                                                                            
  |=========================================                             |  59%
  |                                                                            
  |==========================================                            |  60%
  |                                                                            
  |===========================================                           |  61%
  |                                                                            
  |===========================================                           |  62%
  |                                                                            
  |============================================                          |  62%
  |                                                                            
  |============================================                          |  63%
  |                                                                            
  |=============================================                         |  64%
  |                                                                            
  |==============================================                        |  65%
  |                                                                            
  |==============================================                        |  66%
  |                                                                            
  |===============================================                       |  67%
  |                                                                            
  |===============================================                       |  68%
  |                                                                            
  |================================================                      |  68%
  |                                                                            
  |================================================                      |  69%
  |                                                                            
  |=================================================                     |  69%
  |                                                                            
  |=================================================                     |  70%
  |                                                                            
  |=================================================                     |  71%
  |                                                                            
  |==================================================                    |  71%
  |                                                                            
  |==================================================                    |  72%
  |                                                                            
  |===================================================                   |  72%
  |                                                                            
  |===================================================                   |  73%
  |                                                                            
  |====================================================                  |  74%
  |                                                                            
  |=====================================================                 |  75%
  |                                                                            
  |=====================================================                 |  76%
  |                                                                            
  |======================================================                |  77%
  |                                                                            
  |======================================================                |  78%
  |                                                                            
  |=======================================================               |  78%
  |                                                                            
  |=======================================================               |  79%
  |                                                                            
  |========================================================              |  79%
  |                                                                            
  |========================================================              |  80%
  |                                                                            
  |========================================================              |  81%
  |                                                                            
  |=========================================================             |  81%
  |                                                                            
  |=========================================================             |  82%
  |                                                                            
  |==========================================================            |  82%
  |                                                                            
  |==========================================================            |  83%
  |                                                                            
  |===========================================================           |  84%
  |                                                                            
  |===========================================================           |  85%
  |                                                                            
  |============================================================          |  86%
  |                                                                            
  |=============================================================         |  86%
  |                                                                            
  |=============================================================         |  87%
  |                                                                            
  |==============================================================        |  88%
  |                                                                            
  |==============================================================        |  89%
  |                                                                            
  |===============================================================       |  90%
  |                                                                            
  |================================================================      |  91%
  |                                                                            
  |================================================================      |  92%
  |                                                                            
  |=================================================================     |  92%
  |                                                                            
  |=================================================================     |  93%
  |                                                                            
  |=================================================================     |  94%
  |                                                                            
  |==================================================================    |  94%
  |                                                                            
  |==================================================================    |  95%
  |                                                                            
  |===================================================================   |  95%
  |                                                                            
  |===================================================================   |  96%
  |                                                                            
  |====================================================================  |  97%
  |                                                                            
  |====================================================================  |  98%
  |                                                                            
  |===================================================================== |  98%
  |                                                                            
  |===================================================================== |  99%
  |                                                                            
  |======================================================================| 100%
## 
indexed 2.79kB in  0s, 316.50MB/s
                                                                              
indexed 1.00TB in  0s, 17.18TB/s
                                                                              </code></pre>
<pre><code>## ============================================================================</code></pre>
<pre><code>##  For more information about MAF data please read the following GDC manual and web pages:</code></pre>
<pre><code>##  GDC manual: https://gdc-docs.nci.nih.gov/Data/PDF/Data_UG.pdf</code></pre>
<pre><code>##  https://gdc-docs.nci.nih.gov/Data/Bioinformatics_Pipelines/DNA_Seq_Variant_Calling_Pipeline/</code></pre>
<pre><code>##  https://gdc.cancer.gov/about-gdc/variant-calling-gdc</code></pre>
<pre><code>## ============================================================================</code></pre>
<pre><code>## --------------------------------------</code></pre>
<pre><code>## o GDCquery: Searching in GDC database</code></pre>
<pre><code>## --------------------------------------</code></pre>
<pre><code>## Genome of reference: hg38</code></pre>
<pre><code>## --------------------------------------------</code></pre>
<pre><code>## oo Accessing GDC. This might take a while...</code></pre>
<pre><code>## --------------------------------------------</code></pre>
<pre><code>## ooo Project: TCGA-UVM</code></pre>
<pre><code>## --------------------</code></pre>
<pre><code>## oo Filtering results</code></pre>
<pre><code>## --------------------</code></pre>
<pre><code>## ooo By access</code></pre>
<pre><code>## ooo By data.type</code></pre>
<pre><code>## ooo By workflow.type</code></pre>
<pre><code>## ----------------</code></pre>
<pre><code>## oo Checking data</code></pre>
<pre><code>## ----------------</code></pre>
<pre><code>## ooo Check if there are duplicated cases</code></pre>
<pre><code>## ooo Check if there results for the query</code></pre>
<pre><code>## -------------------</code></pre>
<pre><code>## o Preparing output</code></pre>
<pre><code>## -------------------</code></pre>
<pre><code>## Downloading data for project TCGA-UVM</code></pre>
<pre><code>## GDCdownload will download: 505.413 KB</code></pre>
<pre><code>## Downloading as: TCGA.UVM.mutect.6c7b01bc-b068-4e01-8b4d-0362f5959f65.DR-10.0.somatic.maf.gz</code></pre>
<pre><code>## 
  |                                                                            
  |                                                                      |   0%
  |                                                                            
  |=                                                                     |   2%
  |                                                                            
  |==                                                                    |   3%
  |                                                                            
  |===                                                                   |   5%
  |                                                                            
  |====                                                                  |   6%
  |                                                                            
  |=====                                                                 |   8%
  |                                                                            
  |=======                                                               |   9%
  |                                                                            
  |========                                                              |  11%
  |                                                                            
  |=========                                                             |  13%
  |                                                                            
  |==========                                                            |  14%
  |                                                                            
  |===========                                                           |  16%
  |                                                                            
  |============                                                          |  17%
  |                                                                            
  |=============                                                         |  19%
  |                                                                            
  |==============                                                        |  21%
  |                                                                            
  |===============                                                       |  22%
  |                                                                            
  |=================                                                     |  24%
  |                                                                            
  |==================                                                    |  25%
  |                                                                            
  |===================                                                   |  27%
  |                                                                            
  |====================                                                  |  28%
  |                                                                            
  |=====================                                                 |  30%
  |                                                                            
  |======================                                                |  32%
  |                                                                            
  |=======================                                               |  33%
  |                                                                            
  |========================                                              |  35%
  |                                                                            
  |=========================                                             |  36%
  |                                                                            
  |===========================                                           |  38%
  |                                                                            
  |============================                                          |  40%
  |                                                                            
  |=============================                                         |  41%
  |                                                                            
  |==============================                                        |  43%
  |                                                                            
  |===============================                                       |  44%
  |                                                                            
  |================================                                      |  46%
  |                                                                            
  |=================================                                     |  47%
  |                                                                            
  |==================================                                    |  49%
  |                                                                            
  |===================================                                   |  51%
  |                                                                            
  |=====================================                                 |  52%
  |                                                                            
  |======================================                                |  54%
  |                                                                            
  |=======================================                               |  55%
  |                                                                            
  |========================================                              |  57%
  |                                                                            
  |=========================================                             |  58%
  |                                                                            
  |==========================================                            |  60%
  |                                                                            
  |===========================================                           |  62%
  |                                                                            
  |============================================                          |  63%
  |                                                                            
  |=============================================                         |  65%
  |                                                                            
  |==============================================                        |  66%
  |                                                                            
  |================================================                      |  68%
  |                                                                            
  |=================================================                     |  70%
  |                                                                            
  |==================================================                    |  71%
  |                                                                            
  |===================================================                   |  73%
  |                                                                            
  |====================================================                  |  74%
  |                                                                            
  |=====================================================                 |  76%
  |                                                                            
  |======================================================                |  77%
  |                                                                            
  |=======================================================               |  79%
  |                                                                            
  |========================================================              |  81%
  |                                                                            
  |==========================================================            |  82%
  |                                                                            
  |===========================================================           |  84%
  |                                                                            
  |============================================================          |  85%
  |                                                                            
  |=============================================================         |  87%
  |                                                                            
  |==============================================================        |  89%
  |                                                                            
  |===============================================================       |  90%
  |                                                                            
  |================================================================      |  92%
  |                                                                            
  |=================================================================     |  93%
  |                                                                            
  |==================================================================    |  95%
  |                                                                            
  |====================================================================  |  96%
  |                                                                            
  |===================================================================== |  98%
  |                                                                            
  |======================================================================| 100%
## 
indexed 2.44kB in  0s, 353.48MB/s
                                                                              
indexed 1.00TB in  0s, 83.11TB/s
                                                                              </code></pre>
</div>
<div id="download-clinical-data-using-tcgabiolinks" class="section level3">
<h3 class="hasAnchor">
<a href="#download-clinical-data-using-tcgabiolinks" class="anchor"></a>Download clinical data using <code>TCGABiolinks</code>
</h3>
<div class="sourceCode" id="cb109"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># tcga_clinical &lt;- getTCGAClinicalAnnotation#TCGAbiolinks::GDCquery_clinic(project = paste0("TCGA-",tcga_code), type = "clinical")</span>
<span class="co"># tcga_clinical$Tumor_Sample_Barcode &lt;- tcga_clinical$submitter_id</span>
<span class="va">tcga_clinical</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/getTCGAClinicalAnnotation.html">getTCGAClinicalAnnotation</a></span><span class="op">(</span>cancerCodes <span class="op">=</span> <span class="va">tcga_code</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="make-a-customized-oncoplot" class="section level2">
<h2 class="hasAnchor">
<a href="#make-a-customized-oncoplot" class="anchor"></a>Make a customized oncoplot</h2>
<div id="filter-data" class="section level3">
<h3 class="hasAnchor">
<a href="#filter-data" class="anchor"></a>Filter data</h3>
<p>The <code>filterMAF</code> function can be used to filter the MAF data in various ways. Importantly, by default, it will remove commonly occurring mutations that are often considered to be false position ( <a href="https://bmcmedgenomics.biomedcentral.com/articles/10.1186/s12920-014-0064-y">FLAG genes</a> )</p>
<div class="sourceCode" id="cb110"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">filtered_mafdata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/do.call.html">do.call</a></span><span class="op">(</span><span class="st">"rbind"</span>,<span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">maf_files</span>, <span class="kw">function</span><span class="op">(</span><span class="va">maf_file</span><span class="op">)</span><span class="op">{</span><span class="fu"><a href="../reference/filter_maf_chunked.html">filter_maf_chunked</a></span><span class="op">(</span><span class="va">maf_file</span><span class="op">)</span><span class="op">}</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div id="add-clinical-data" class="section level3">
<h3 class="hasAnchor">
<a href="#add-clinical-data" class="anchor"></a>Add clinical data</h3>
<p>The easiest way to add clinical annotations to the oncoplot is to add clinical data to the <code>clinical.data</code> slot of a <code>MAF</code> object before passing it to the <code>generateOncoplot()</code> function.</p>
<p>MAFDash also provides a function that defines reasonable colors for some common clinical annotations provided with TCGA datasets.</p>
<div class="sourceCode" id="cb111"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">filtered_maf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/maftools/man/read.maf.html">read.maf</a></span><span class="op">(</span><span class="va">filtered_mafdata</span>, clinicalData <span class="op">=</span> <span class="va">tcga_clinical</span><span class="op">$</span><span class="va">annodata</span><span class="op">)</span></code></pre></div>
<pre><code>## -Validating
## -Silent variants: 4326 
## -Summarizing
## --Mutiple centers found
## BCM;BI--Possible FLAGS among top ten genes:
##   MUC16
##   TTN
## -Processing clinical data
## -Finished in 0.668s elapsed (0.632s cpu)</code></pre>
<div class="sourceCode" id="cb113"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">annotation_colors</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getTCGAClinicalColors.html">getTCGAClinicalColors</a></span><span class="op">(</span>ageRange <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/range.html">range</a></span><span class="op">(</span><span class="va">tcga_clinical</span><span class="op">$</span><span class="va">annodata</span><span class="op">$</span><span class="va">age_at_diagnosis</span>, na.rm<span class="op">=</span><span class="cn">T</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div id="make-an-annotated-oncoplot" class="section level3">
<h3 class="hasAnchor">
<a href="#make-an-annotated-oncoplot" class="anchor"></a>Make an annotated oncoplot</h3>
<p>The <code>add_clinical_annotations</code> argument can be:</p>
<ul>
<li>A boolean indicating whether or not to add annotations built from the <code>clinical.data</code> slot of the <code>MAF</code> object. Columns with all missing values are ignored. Maximum number of annotations plotted is 10 (first 10 non-empty columns of <code>clinical.data</code>)</li>
<li>A character vector of column names provided as clinical data</li>
</ul>
<div class="sourceCode" id="cb114"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">custom_onco</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/generateOncoPlot.html">generateOncoPlot</a></span><span class="op">(</span><span class="va">filtered_maf</span>, 
                                add_clinical_annotations <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">annotation_colors</span><span class="op">)</span>, 
                                clin_data_colors <span class="op">=</span> <span class="va">annotation_colors</span><span class="op">)</span></code></pre></div>
<pre><code>## All mutation types: Missense Mutation, Frame Shift Del, Splice Site,
## Multi Hit, Nonsense Mutation, In Frame Del, Frame Shift Ins.</code></pre>
<pre><code>## `alter_fun` is assumed vectorizable. If it does not generate correct
## plot, please set `alter_fun_is_vectorized = FALSE` in `oncoPrint()`.</code></pre>
</div>
</div>
<div id="make-some-other-figures" class="section level2">
<h2 class="hasAnchor">
<a href="#make-some-other-figures" class="anchor"></a>Make some other figures</h2>
<div id="tcga-comparison" class="section level3">
<h3 class="hasAnchor">
<a href="#tcga-comparison" class="anchor"></a>TCGA Comparison</h3>
<p>A lot of <code>maftools</code>’s plots are base graphics, so they’re drawn to a device and not returned. But we can simply save them to a file and provide the file path.</p>
<div class="sourceCode" id="cb117"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">tcgaComparePlot</span><span class="op">&lt;-</span><span class="fu"><a href="../reference/generateTCGAComparePlot.html">generateTCGAComparePlot</a></span><span class="op">(</span>maf <span class="op">=</span> <span class="va">filtered_maf</span>, cohortName <span class="op">=</span> <span class="st">"test"</span><span class="op">)</span></code></pre></div>
<pre><code>## Warning in FUN(X[[i]], ...): Removed 0 samples with zero mutations.</code></pre>
<pre><code>## Performing pairwise t-test for differences in mutation burden..</code></pre>
</div>
<div id="chord-diagram-of-mutation-co-occurrence" class="section level3">
<h3 class="hasAnchor">
<a href="#chord-diagram-of-mutation-co-occurrence" class="anchor"></a>Chord Diagram of mutation co-occurrence</h3>
<p>This function is built on top of <code>maftools</code>’s <code><a href="https://rdrr.io/pkg/maftools/man/somaticInteractions.html">somaticInteractions()</a></code> function. It’s just a different way of visualizing co-occurence or mutual exclusivity between genes.</p>
<div class="sourceCode" id="cb120"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ribbonplot_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/getwd.html">getwd</a></span><span class="op">(</span><span class="op">)</span>,<span class="st">"ribbon.pdf"</span><span class="op">)</span>
<span class="fu"><a href="../reference/generateRibbonPlot.html">generateRibbonPlot</a></span><span class="op">(</span><span class="va">filtered_maf</span>,save_name <span class="op">=</span> <span class="va">ribbonplot_file</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div id="render-the-dashboard" class="section level2">
<h2 class="hasAnchor">
<a href="#render-the-dashboard" class="anchor"></a>Render the dashboard</h2>
<div class="sourceCode" id="cb121"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">customplotlist</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="st">"summary_plot"</span><span class="op">=</span><span class="cn">T</span>,
                       <span class="st">"burden"</span><span class="op">=</span><span class="cn">T</span>,
                       <span class="st">"TCGA Comparison"</span><span class="op">=</span><span class="va">tcgaComparePlot</span><span class="op">$</span><span class="va">tcga_compare_plot</span>,
                       <span class="st">"oncoplot"</span><span class="op">=</span><span class="cn">T</span>,
                       <span class="st">"Annotated Oncoplot"</span><span class="op">=</span><span class="va">custom_onco</span>
                       <span class="op">)</span>

<span class="co">## Filename to output to; if output directory doesn't exist, it will be created</span>
<span class="va">html_filename</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/tempfile.html">tempdir</a></span><span class="op">(</span><span class="op">)</span>,<span class="st">"/TCGA-UVM.custom.mafdash.html"</span><span class="op">)</span><span class="op">)</span>

<span class="co">## Render dashboard</span>
<span class="fu"><a href="../reference/getMAFDashboard.html">getMAFDashboard</a></span><span class="op">(</span>MAFfilePath <span class="op">=</span> <span class="va">filtered_maf</span>,
                plotList <span class="op">=</span> <span class="va">customplotlist</span>,
                outputFileName <span class="op">=</span> <span class="va">html_filename</span>, 
                outputFileTitle <span class="op">=</span> <span class="st">"Customized Dashboard"</span><span class="op">)</span></code></pre></div>
</div>
<div id="output" class="section level2">
<h2 class="hasAnchor">
<a href="#output" class="anchor"></a>Output</h2>
<p>The output can be seen <a href="https://mtandon09.github.io/MAFDashRPackage/examples/TCGA-UVM.custom.mafdash.html.MAFDash.html">here</a>.</p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="https://github.com/ashishjain1988">Ashish Jain</a>, <a href="https://github.com/mtandon09">Mayank Tandon</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
