<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Computes and plot the cosine similarity of each individual signature is computed against each COSMIC signature
from COMIC V3.2 — generateCOSMICMutSigSimHeatmap • MAFDash</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Computes and plot the cosine similarity of each individual signature is computed against each COSMIC signature
from COMIC V3.2 — generateCOSMICMutSigSimHeatmap"><meta property="og:description" content="This function computes and plot the cosine similarity of each individual signature is computed against each COSMIC signature
from COMIC V3.2. The cosine similarity is a value between 0 (distinct) and 1 (identical) and indicates how much two vectors
are alike."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MAFDash</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.2.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/MAFDash.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/Advanced_Example.html">Advanced Example</a>
    </li>
    <li>
      <a href="../articles/Customizing.html">Customizing</a>
    </li>
    <li>
      <a href="../articles/Quick_Start.html">Quick Start</a>
    </li>
  </ul></li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/ashishjain1988/MAFDash/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Computes and plot the cosine similarity of each individual signature is computed against each COSMIC signature
from COMIC V3.2</h1>
    <small class="dont-index">Source: <a href="https://github.com/ashishjain1988/MAFDash/blob/HEAD/R/generateCOSMICSignaturePlot.R" class="external-link"><code>R/generateCOSMICSignaturePlot.R</code></a></small>
    <div class="hidden name"><code>generateCOSMICMutSigSimHeatmap.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function computes and plot the cosine similarity of each individual signature is computed against each COSMIC signature
from COMIC V3.2. The cosine similarity is a value between 0 (distinct) and 1 (identical) and indicates how much two vectors
are alike.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">generateCOSMICMutSigSimHeatmap</span><span class="op">(</span>
  <span class="va">mymaf</span>,
  use_silent_mutations <span class="op">=</span> <span class="cn">FALSE</span>,
  full_output <span class="op">=</span> <span class="cn">FALSE</span>,
  show_broad_categories <span class="op">=</span> <span class="cn">TRUE</span>,
  clin_data <span class="op">=</span> <span class="cn">NULL</span>,
  clin_data_colors <span class="op">=</span> <span class="cn">NULL</span>,
  add_sample_names <span class="op">=</span> <span class="cn">NULL</span>,
  savename <span class="op">=</span> <span class="cn">NULL</span>,
  fig_height <span class="op">=</span> <span class="cn">NULL</span>,
  fig_width <span class="op">=</span> <span class="cn">NULL</span>
<span class="op">)</span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>mymaf</dt>
<dd><p>mutation count matrix (dimensions: a mutation features X n samples)</p></dd>
<dt>use_silent_mutations</dt>
<dd><p>96 mutation count matrix (dimensions: a mutation features X m samples)</p></dd>
<dt>full_output</dt>
<dd><p>return full output including the etiology matrix and plot data</p></dd>
<dt>show_broad_categories</dt>
<dd><p>To show broad etiology categories</p></dd>
<dt>clin_data</dt>
<dd><p>Clinical data to be plotted in the heatmap</p></dd>
<dt>clin_data_colors</dt>
<dd><p>Clinical data colors</p></dd>
<dt>add_sample_names</dt>
<dd><p>Whether or not to add column labels; if set to NULL, will add labels only if # samples less than 10</p></dd>
<dt>savename</dt>
<dd><p>file name of the plot</p></dd>
<dt>fig_height</dt>
<dd><p>Output height (inches); set to NULL to size automatically; only used if savename is set.</p></dd>
<dt>fig_width</dt>
<dd><p>Output width (inches); set to NULL to size automatically; only used if savename is set.</p></dd>
</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>Complex Heatmap object. If full_output is TRUE it will consist of a list including heatmap object,
etiology matrix and plot data.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/ashishjain1988/MAFDash/" class="external-link">MAFDash</a></span><span class="op">)</span></span>
<span class="r-in"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/PoisonAlien/maftools" class="external-link">maftools</a></span><span class="op">)</span></span>
<span class="r-in"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/jokergoo/ComplexHeatmap" class="external-link">ComplexHeatmap</a></span><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: grid</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========================================</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ComplexHeatmap version 2.6.2</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Bioconductor page: http://bioconductor.org/packages/ComplexHeatmap/</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Github page: https://github.com/jokergoo/ComplexHeatmap</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Documentation: http://jokergoo.github.io/ComplexHeatmap-reference</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> If you use it in published research, please cite:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Gu, Z. Complex heatmaps reveal patterns and correlations in multidimensional </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   genomic data. Bioinformatics 2016.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> This message can be suppressed by:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   suppressPackageStartupMessages(library(ComplexHeatmap))</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> ========================================</span>
<span class="r-in"><span class="va">maf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"test.mutect2.maf.gz"</span>, package <span class="op">=</span> <span class="st">"MAFDash"</span><span class="op">)</span></span>
<span class="r-in"><span class="va">val</span><span class="op">&lt;-</span><span class="fu">generateCOSMICMutSigSimHeatmap</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/maftools/man/read.maf.html" class="external-link">read.maf</a></span><span class="op">(</span><span class="va">maf</span><span class="op">)</span><span class="op">)</span>;<span class="fu"><a href="https://rdrr.io/pkg/ComplexHeatmap/man/draw-dispatch.html" class="external-link">draw</a></span><span class="op">(</span><span class="va">val</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> -Reading</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> -Validating</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> -Silent variants: 561 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> -Summarizing</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --Possible FLAGS among top ten genes:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   MACF1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   MUC16</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> -Processing clinical data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --Missing clinical data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> -Finished in 0.244s elapsed (0.220s cpu) </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: ‘BiocGenerics’</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following objects are masked from ‘package:parallel’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     clusterExport, clusterMap, parApply, parCapply, parLapply,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     parLapplyLB, parRapply, parSapply, parSapplyLB</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following objects are masked from ‘package:stats’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     IQR, mad, sd, var, xtabs</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following objects are masked from ‘package:base’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Filter, Find, Map, Position, Reduce, anyDuplicated, append,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     as.data.frame, basename, cbind, colnames, dirname, do.call,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     duplicated, eval, evalq, get, grep, grepl, intersect, is.unsorted,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     lapply, mapply, match, mget, order, paste, pmax, pmax.int, pmin,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     pmin.int, rank, rbind, rownames, sapply, setdiff, sort, table,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     tapply, union, unique, unsplit, which.max, which.min</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: ‘S4Vectors’</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following object is masked from ‘package:base’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     expand.grid</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: ‘Biostrings’</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following object is masked from ‘package:base’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     strsplit</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> -Extracting 5' and 3' adjacent bases</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> -Extracting +/- 20bp around mutated bases for background C&gt;T estimation</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> -Estimating APOBEC enrichment scores</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --Performing one-way Fisher's test for APOBEC enrichment</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ---APOBEC related mutations are enriched in  12.658 % of samples (APOBEC enrichment score &gt; 2 ;  10  of  79  samples)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> -Creating mutation matrix</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --matrix of dimension 80x96</span>
<span class="r-plt img"><img src="generateCOSMICMutSigSimHeatmap-1.png" alt="" width="700" height="433"></span>
<span class="r-in"></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Ashish Jain, Mayank Tandon.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer></div>

  


  

  </body></html>

