<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Plot the comparison of the mutation load against TCGA cohorts — generateTCGAComparePlot • MAFDash</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Plot the comparison of the mutation load against TCGA cohorts — generateTCGAComparePlot"><meta property="og:description" content="This function computes and plot the mutation load of the input MAF against all of 33 TCGA cohorts derived from MC3 project."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MAFDash</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.2.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/MAFDash.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/Advanced_Example.html">Advanced Example</a>
    </li>
    <li>
      <a href="../articles/Customizing.html">Customizing</a>
    </li>
    <li>
      <a href="../articles/Quick_Start.html">Quick Start</a>
    </li>
  </ul></li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/ashishjain1988/MAFDash/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Plot the comparison of the mutation load against TCGA cohorts</h1>
    <small class="dont-index">Source: <a href="https://github.com/ashishjain1988/MAFDash/blob/HEAD/R/generateTCGAComparePlot.R" class="external-link"><code>R/generateTCGAComparePlot.R</code></a></small>
    <div class="hidden name"><code>generateTCGAComparePlot.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function computes and plot the mutation load of the input MAF against all of 33 TCGA cohorts derived from MC3 project.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">generateTCGAComparePlot</span><span class="op">(</span>
  <span class="va">maf</span>,
  capture_size <span class="op">=</span> <span class="cn">NULL</span>,
  tcga_capture_size <span class="op">=</span> <span class="fl">35.8</span>,
  cohortName <span class="op">=</span> <span class="cn">NULL</span>,
  tcga_cohorts <span class="op">=</span> <span class="cn">NULL</span>,
  primarySite <span class="op">=</span> <span class="cn">FALSE</span>,
  col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"gray70"</span>, <span class="st">"black"</span><span class="op">)</span>,
  medianCol <span class="op">=</span> <span class="st">"red"</span>,
  decreasing <span class="op">=</span> <span class="cn">FALSE</span>,
  logscale <span class="op">=</span> <span class="cn">TRUE</span>,
  rm_hyper <span class="op">=</span> <span class="cn">FALSE</span>,
  rm_zero <span class="op">=</span> <span class="cn">TRUE</span>
<span class="op">)</span></code></pre></div>
    </div>

    <div id="source">
    <h2>Source</h2>
    <p>TCGA MC3 file was obtained from  https://api.gdc.cancer.gov/data/1c8cfe5f-e52d-41ba-94da-f15ea1337efc. See TCGAmutations R package for more details. Further downstream script to estimate TMB for each sample can be found in ‘inst/scripts/estimate_tcga_tmb.R’</p>
    </div>
    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>maf</dt>
<dd><p>A MAF object</p></dd>
<dt>capture_size</dt>
<dd><p>capture size for input MAF in MBs. Default NULL. If provided plot will be scaled to mutations per MB</p></dd>
<dt>tcga_capture_size</dt>
<dd><p>capture size for TCGA cohort in MB. Default 35.8.</p></dd>
<dt>cohortName</dt>
<dd><p>name for the input MAF cohort. Default "Input"</p></dd>
<dt>tcga_cohorts</dt>
<dd><p>restrict tcga data to these cohorts.</p></dd>
<dt>primarySite</dt>
<dd><p>If TRUE uses primary site of cancer as labels instead of TCGA project IDs. Default FALSE.</p></dd>
<dt>col</dt>
<dd><p>color vector for length 2 TCGA cohorts and input MAF cohort. Default gray70 and black.</p></dd>
<dt>medianCol</dt>
<dd><p>color for median line. Default red.</p></dd>
<dt>decreasing</dt>
<dd><p>Default FALSE. Cohorts are arranged in increasing mutation burden.</p></dd>
<dt>logscale</dt>
<dd><p>Default TRUE</p></dd>
<dt>rm_hyper</dt>
<dd><p>Remove hyper mutated samples (outliers)? Default FALSE</p></dd>
<dt>rm_zero</dt>
<dd><p>Remove samples with zero mutations? Default TRUE</p></dd>
</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>A list consisting of median mutations per cohort, mutation burden per cohort,
significant differences between cohorts, and ggplot object to show mutation burden plot</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>Tumor mutation burden for TCGA cohorts is obtained from TCGA MC3 study. For consistency TMB is estimated by restricting variants within Agilent Sureselect capture kit of size 35.8 MB.</p>
    </div>
    <div id="references">
    <h2>References</h2>
    <p>Scalable Open Science Approach for Mutation Calling of Tumor Exomes Using Multiple Genomic Pipelines Kyle Ellrott, Matthew H. Bailey, Gordon Saksena, et. al. Cell Syst. 2018 Mar 28; 6(3): 271–281.e7. https://doi.org/10.1016/j.cels.2018.03.002</p>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Ashish Jain, Mayank Tandon</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/PoisonAlien/maftools" class="external-link">maftools</a></span><span class="op">)</span></span>
<span class="r-in"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/ashishjain1988/MAFDash/" class="external-link">MAFDash</a></span><span class="op">)</span></span>
<span class="r-in"><span class="va">maf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"test.mutect2.maf.gz"</span>, package <span class="op">=</span> <span class="st">"MAFDash"</span><span class="op">)</span></span>
<span class="r-in"><span class="va">maf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/maftools/man/read.maf.html" class="external-link">read.maf</a></span><span class="op">(</span>maf <span class="op">=</span> <span class="va">maf</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> -Reading</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> -Validating</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> -Silent variants: 561 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> -Summarizing</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --Possible FLAGS among top ten genes:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   MACF1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   MUC16</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> -Processing clinical data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --Missing clinical data</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> -Finished in 0.214s elapsed (0.195s cpu) </span>
<span class="r-in"><span class="va">l</span><span class="op">&lt;-</span><span class="fu">generateTCGAComparePlot</span><span class="op">(</span>maf <span class="op">=</span> <span class="va">maf</span>, cohortName <span class="op">=</span> <span class="st">"test"</span><span class="op">)</span></span>
<span class="r-in"><span class="va">l</span><span class="op">$</span><span class="va">tcga_compare_plot</span></span>
<span class="r-plt img"><img src="generateTCGAComparePlot-1.png" alt="" width="700" height="433"></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Ashish Jain, Mayank Tandon.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer></div>

  


  

  </body></html>

