<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Computes and plot the cosine similarity of each individual signature is computed against each COSMIC signature
from COMIC V3.2 — generateCOSMICMutSigSimHeatmap • MAFDash</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Computes and plot the cosine similarity of each individual signature is computed against each COSMIC signature
from COMIC V3.2 — generateCOSMICMutSigSimHeatmap" />
<meta property="og:description" content="This function computes and plot the cosine similarity of each individual signature is computed against each COSMIC signature
from COMIC V3.2. The cosine similarity is a value between 0 (distinct) and 1 (identical) and indicates how much two vectors
are alike." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">MAFDash</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.2.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/MAFDash.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/Advanced_Example.html">Advanced Example</a>
    </li>
    <li>
      <a href="../articles/Customizing.html">Customizing</a>
    </li>
    <li>
      <a href="../articles/Quick_Start.html">Quick Start</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/ashishjain1988/MAFDash/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Computes and plot the cosine similarity of each individual signature is computed against each COSMIC signature
from COMIC V3.2</h1>
    <small class="dont-index">Source: <a href='https://github.com/ashishjain1988/MAFDash/blob/master/R/generateCOSMICSignaturePlot.R'><code>R/generateCOSMICSignaturePlot.R</code></a></small>
    <div class="hidden name"><code>generateCOSMICMutSigSimHeatmap.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function computes and plot the cosine similarity of each individual signature is computed against each COSMIC signature
from COMIC V3.2. The cosine similarity is a value between 0 (distinct) and 1 (identical) and indicates how much two vectors
are alike.</p>
    </div>

    <pre class="usage"><span class='fu'>generateCOSMICMutSigSimHeatmap</span><span class='op'>(</span>
  <span class='va'>mymaf</span>,
  use_silent_mutations <span class='op'>=</span> <span class='cn'>FALSE</span>,
  full_output <span class='op'>=</span> <span class='cn'>FALSE</span>,
  show_broad_categories <span class='op'>=</span> <span class='cn'>TRUE</span>,
  clin_data <span class='op'>=</span> <span class='cn'>NULL</span>,
  clin_data_colors <span class='op'>=</span> <span class='cn'>NULL</span>,
  add_sample_names <span class='op'>=</span> <span class='cn'>NULL</span>,
  savename <span class='op'>=</span> <span class='cn'>NULL</span>,
  fig_height <span class='op'>=</span> <span class='cn'>NULL</span>,
  fig_width <span class='op'>=</span> <span class='cn'>NULL</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>mymaf</th>
      <td><p>mutation count matrix (dimensions: a mutation features X n samples)</p></td>
    </tr>
    <tr>
      <th>use_silent_mutations</th>
      <td><p>96 mutation count matrix (dimensions: a mutation features X m samples)</p></td>
    </tr>
    <tr>
      <th>full_output</th>
      <td><p>return full output including the etiology matrix and plot data</p></td>
    </tr>
    <tr>
      <th>show_broad_categories</th>
      <td><p>To show broad etiology categories</p></td>
    </tr>
    <tr>
      <th>clin_data</th>
      <td><p>Clinical data to be plotted in the heatmap</p></td>
    </tr>
    <tr>
      <th>clin_data_colors</th>
      <td><p>Clinical data colors</p></td>
    </tr>
    <tr>
      <th>add_sample_names</th>
      <td><p>Whether or not to add column labels; if set to NULL, will add labels only if # samples less than 10</p></td>
    </tr>
    <tr>
      <th>savename</th>
      <td><p>file name of the plot</p></td>
    </tr>
    <tr>
      <th>fig_height</th>
      <td><p>Output height (inches); set to NULL to size automatically; only used if savename is set.</p></td>
    </tr>
    <tr>
      <th>fig_width</th>
      <td><p>Output width (inches); set to NULL to size automatically; only used if savename is set.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>Complex Heatmap object. If full_output is TRUE it will consist of a list including heatmap object,
etiology matrix and plot data.</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://github.com/ashishjain1988/MAFDash/'>MAFDash</a></span><span class='op'>)</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://github.com/PoisonAlien/maftools'>maftools</a></span><span class='op'>)</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='https://github.com/jokergoo/ComplexHeatmap'>ComplexHeatmap</a></span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Loading required package: grid</span></div><div class='output co'>#&gt; <span class='message'>========================================</span>
#&gt; <span class='message'>ComplexHeatmap version 2.6.2</span>
#&gt; <span class='message'>Bioconductor page: http://bioconductor.org/packages/ComplexHeatmap/</span>
#&gt; <span class='message'>Github page: https://github.com/jokergoo/ComplexHeatmap</span>
#&gt; <span class='message'>Documentation: http://jokergoo.github.io/ComplexHeatmap-reference</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>If you use it in published research, please cite:</span>
#&gt; <span class='message'>Gu, Z. Complex heatmaps reveal patterns and correlations in multidimensional </span>
#&gt; <span class='message'>  genomic data. Bioinformatics 2016.</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>This message can be suppressed by:</span>
#&gt; <span class='message'>  suppressPackageStartupMessages(library(ComplexHeatmap))</span>
#&gt; <span class='message'>========================================</span></div><div class='input'><span class='va'>maf</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span><span class='op'>(</span><span class='st'>"extdata"</span>, <span class='st'>"test.mutect2.maf.gz"</span>, package <span class='op'>=</span> <span class='st'>"MAFDash"</span><span class='op'>)</span>
<span class='va'>val</span><span class='op'>&lt;-</span><span class='fu'>generateCOSMICMutSigSimHeatmap</span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/pkg/maftools/man/read.maf.html'>read.maf</a></span><span class='op'>(</span><span class='va'>maf</span><span class='op'>)</span><span class='op'>)</span>;<span class='fu'><a href='https://rdrr.io/pkg/ComplexHeatmap/man/draw-dispatch.html'>draw</a></span><span class='op'>(</span><span class='va'>val</span><span class='op'>)</span>
</div><div class='output co'>#&gt; -Reading
#&gt; -Validating
#&gt; -Silent variants: 561 
#&gt; -Summarizing
#&gt; --Possible FLAGS among top ten genes:
#&gt;   MACF1
#&gt;   MUC16
#&gt; -Processing clinical data
#&gt; --Missing clinical data
#&gt; -Finished in 0.261s elapsed (0.235s cpu) </div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: ‘BiocGenerics’</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from ‘package:parallel’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,</span>
#&gt; <span class='message'>    clusterExport, clusterMap, parApply, parCapply, parLapply,</span>
#&gt; <span class='message'>    parLapplyLB, parRapply, parSapply, parSapplyLB</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from ‘package:stats’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    IQR, mad, sd, var, xtabs</span></div><div class='output co'>#&gt; <span class='message'>The following objects are masked from ‘package:base’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    anyDuplicated, append, as.data.frame, basename, cbind, colnames,</span>
#&gt; <span class='message'>    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,</span>
#&gt; <span class='message'>    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,</span>
#&gt; <span class='message'>    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,</span>
#&gt; <span class='message'>    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,</span>
#&gt; <span class='message'>    union, unique, unsplit, which.max, which.min</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: ‘S4Vectors’</span></div><div class='output co'>#&gt; <span class='message'>The following object is masked from ‘package:base’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    expand.grid</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: ‘Biostrings’</span></div><div class='output co'>#&gt; <span class='message'>The following object is masked from ‘package:base’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    strsplit</span></div><div class='output co'>#&gt; -Extracting 5' and 3' adjacent bases
#&gt; -Extracting +/- 20bp around mutated bases for background C&gt;T estimation
#&gt; -Estimating APOBEC enrichment scores
#&gt; --Performing one-way Fisher's test for APOBEC enrichment
#&gt; ---APOBEC related mutations are enriched in  12.658 % of samples (APOBEC enrichment score &gt; 2 ;  10  of  79  samples)
#&gt; -Creating mutation matrix
#&gt; --matrix of dimension 80x96</div><div class='img'><img src='generateCOSMICMutSigSimHeatmap-1.png' alt='' width='700' height='433' /></div><div class='input'>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Ashish Jain, Mayank Tandon.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


